BEIMA.Backend.Test/MongoConnectorTest.cs

```diff
@@ -0,0 +1,206 @@
+            BsonDocument doc = new BsonDocument {
+                { "deviceTypeId", "a" },
+                { "serialNumber", "a12345"}
+            };
+            var insertResult = mongo.InsertDevice(doc);
+            Assert.IsNotNull(insertResult);
+            Assert.That(insertResult, Is.TypeOf(typeof(ObjectId)));
+
+            //Test (retrieve all documents)
+            var retrievedDocs = mongo.GetAllDevices();
+            Assert.IsNotNull(retrievedDocs);
+            Assert.That(retrievedDocs, Is.TypeOf(typeof(List<BsonDocument>)));
+        }
+
+        [Test]
+        public void DocumentNotInserted_InsertDocument_DocumentHasBeenInserted()
+        {
+            var mongo = MongoConnector.Instance;
+            BsonDocument doc = new BsonDocument { 
+                { "deviceTypeId", "a" },
+                { "serialNumber", "a12345"}
+            };
+            var result = mongo.InsertDevice(doc);
+            Console.WriteLine(result.ToString());
+            Assert.IsNotNull(result);
+            Assert.That(result, Is.TypeOf(typeof(ObjectId)));
+        }
+
+        [Test]
+        public void DocumentNotInserted_InsertDocumentAndDeleteDocument_DocumentHasBeenDeleted()
+        {
+            var mongo = MongoConnector.Instance;
+            BsonDocument doc = new BsonDocument
+            {
+                { "deviceTypeId", "a" },
+                { "serialNumber", "a12345"}
+            };
+            //Insert device
+            var insertResult = mongo.InsertDevice(doc);
+            Assert.IsNotNull(insertResult);
+            Assert.That(insertResult, Is.TypeOf(typeof(ObjectId)));
+
+            //Delete device
+            bool deleteResult = false;
+            if(insertResult != null)
+            {
+                deleteResult = mongo.DeleteDevice((ObjectId) insertResult);
+            }
+            Assert.IsTrue(deleteResult);
+        }
+
+        [Test]
+        public void DocumentNotInserted_InsertDocumentAndUpdateDocument_DocumentHasBeenUpdated()
+        {
+            var mongo = MongoConnector.Instance;
+            BsonDocument doc = new BsonDocument
+            {
+                { "deviceTypeId", "a" },
+                { "serialNumber", "a12345"}
+            };
+            //Insert device
+            var insertResult = mongo.InsertDevice(doc);
+            Assert.IsNotNull(insertResult);
+            Assert.That(insertResult, Is.TypeOf(typeof(ObjectId)));
+
+            //Update device
+            doc.AddRange(new BsonDocument { { "updatedDeviceField", "123" } });
+            var updateResult = mongo.UpdateDevice(doc);
+            Assert.IsTrue(updateResult);
+        }
+
+        [Test]
+        public void ConnectorCreated_GetDeviceGivenInvalidDeviceId_NullReturned()
+        {
+            var mongo = MongoConnector.Instance;
+            //This is a valid ObjectId, but this is not in the database
+            var doc = mongo.GetDevice(new ObjectId("61f4882f42dab933544849d3"));
```

Comment:
Will there be a situation where this could become an actual ID in the database? Are we using a specific test database?

Reply:
I think that the chances of this actually happening with a guid are so miniscule, it shouldn't be an issue. 1 in 2^128 I believe.

Reply:
There shouldn't be, unless we are extremely unlucky. The probability of this occurring is 16^24 I think? I had to use some hexadecimal value of length 24, otherwise ObjectId would refuse to instantiate.

Reply:
Having separate collections solves this issue entirely, be69b94